doc/playbook.rst

```diff
@@ -195,3 +207,73 @@ On a VM with the Science Pipelines installed, a new APDB schema can be created i
 .. code-block:: sh
 
     make_apdb.py -c isolation_level=READ_UNCOMMITTED -c db_url="postgresql://postgres@localhost/postgres"
+
+
+Middleware Worker VM
+====================
+
+The ``rubin-utility-middleware`` VM on Google Compute Engine is intended as a general-purpose environment for working with Butler repositories.
+It can work with both local repositories and ones based on Google Storage.
+
+Built-in support:
+
+* a complete install of the Science Pipelines in ``/software/lsst_stack/``
+* a running instance of ``cloud_sql_proxy`` mapping the ``butler-registry`` database to port 5432
+* global configuration pointing Butler ``s3://`` URIs to Google Storage buckets
+
+The user is responsible for:
+
+* running ``/software/lsst_stack/loadLSST.sh`` on login
+* database authentication (see `Databases`_, above)
+* `Google Storage Authentication`_
+
+
+Google Storage Authentication
+-----------------------------
+
+To access `Google Storage-Backed Repositories`_, you must first set up Boto authentication.
+If you don't have one, `create an HMAC key`_ (this is *not* the same as the token for running the `tester`_); the relevant service account is ``service-620570835826@gs-project-accounts.iam.gserviceaccount.com``.
+Then create a ``~/.aws/credentials`` file with the contents::
+
+    [default]
+    aws_access_key_id=<access key>
+    aws_secret_access_key=<secret key>
+
+and execute ``chmod go-rwx ~/.aws/credentials``.
+
+.. _create an HMAC key: https://cloud.google.com/storage/docs/authentication/managing-hmackeys#create
+
+PostgreSQL-Backed Repositories
+------------------------------
+
+By default, ``butler create`` creates a repository whose registry is stored in SQLite.
+To instead store the registry in the ``butler-registry`` database, create a seed config YAML such as::
+
+    registry:
+      db: postgresql://postgres@localhost:5432/
+      namespace: <unique namespace>
+
+Then run ``butler create --seed-config seedconfig.yaml <repo location>`` to create the repository.
+
+Each repository needs its own ``namespace`` value, corresponding to a PostgreSQL schema.
+Schemas can be listed from within ``psql`` using the ``\dn`` command, and corrupted or outdated registries can be deleted using the ``DROP SCHEMA`` command.
+
+.. warning::
```

Comment:
(I'm not certain what markup you're trying to use here but it doesn't seem to have rendered in the output?)

Reply:
It's a [standard directive](https://docutils.sourceforge.io/docs/ref/rst/directives.html#admonitions); it's unfortunate that GitHub doesn't support it, but it's better than nothing.